pkg/controller/secret/secretsync.go

```diff
@@ -0,0 +1,208 @@
+package secret
+
+import (
+	"context"
+	"fmt"
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/IBM/platform-services-go-sdk/resourcemanagerv2"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/klog/v2"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/IBM/ibm-vpc-block-csi-driver-operator/pkg/util"
+)
+
+// This SecretSyncController translates Secret provided by cloud-credential-operator into
+// format required by the CSI driver.
+type SecretSyncController struct {
+	operatorClient  v1helpers.OperatorClient
+	kubeClient      kubernetes.Interface
+	secretLister    corelisters.SecretLister
+	configMapLister corelisters.ConfigMapLister
+	eventRecorder   events.Recorder
+}
+
+const (
+	// Name of key with ibmcloud_api_key in Secret provided by cloud-credentials-operator.
+	cloudSecretKey = "ibmcloud_api_key"
+	// Name of key with cloud.conf in ConfigMap provided by cloud-credentials-operator.
+	CloudConfigmapKey = "cloud.conf"
+	// Name of the key in secret storage-secret-store creating on operator namespace
+	StorageSecretStoreKey = "slclient.toml"
+
+	// storage-secret-store data format
+	StorageSecretTomlTemplate = `[vpc]
+iam_client_id = "bx"
+iam_client_secret = "bx"
+
+g2_token_exchange_endpoint_url = "https://iam.bluemix.net"
```

Comment:
This IAM endpoint is changed. Please update as per what we have in our odf agent

Reply:
changed